rosbag2_transport/test/rosbag2_transport/test_rosbag2_node.cpp

```diff
@@ -125,6 +135,68 @@ TEST_F(RosBag2NodeFixture, publisher_and_subscriber_work)
   EXPECT_THAT(subscribed_messages[0], StrEq("Hello World"));
 }
 
+TEST_F(RosBag2NodeFixture, generic_subscription_uses_qos)
+{
+  // If the GenericSubscription does not use the provided QoS profile,
+  // its request will be incompatible with the Publisher's offer and no messages will be passed.
+  using namespace std::chrono_literals;
+  std::vector<std::string> test_messages = {"Hello World", "Hello World"};
+  std::string topic_name = "string_topic";
+  std::string type = "test_msgs/Strings";
+  rclcpp::QoS qos = rclcpp::SensorDataQoS();
+  auto publisher = node_->create_publisher<test_msgs::msg::Strings>(topic_name, qos);
+
+  auto subscriber_future_ = std::async(
+    std::launch::async, [this, topic_name, type, qos] {
+      return subscribe_raw_messages(1, topic_name, type, qos);
+    });
+  // Give time to the subscriber to start.
+  std::this_thread::sleep_for(std::chrono::milliseconds(100));
+
+  for (const auto & message : test_messages) {
+    test_msgs::msg::Strings string_message;
+    string_message.string_value = message;
+    publisher->publish(string_message);
+    // This is necessary because sometimes, the subscriber is initialized very late
```

Comment:
Do you smell flaky? Because I smell flaky :)

Reply:
Copying existing test structure from this file which has been working reliably - I'm personally willing to take on the tech debt of "possibly flaky test" right now in order to hit the Foxy release date, over restructuring the tests now. What do you say?

Reply:
It's definitely flaky and part of the reason why some end to end tests are disabled. I am happy to track this in a follow-up issue, but I think it's a rabbit hole we shouldn't go down here in this PR. But that's just my 2cts.

Reply:
I was able to greatly simplify these tests to avoid the sleeps